### PR TITLE
[JS-to-C++ test] Test SCardEstablishContext

### PR DESCRIPTION
**Pull request by emaxx-google**
Add the first JS-to-C++ test: SCardEstablishContext() function from the JavaScript API.

This commit contributes to #869.

### CODE REVIEW

File: smart_card_connector_app/src/pcsc-api-jstocxxtest.js [hunk 1]
```diff
@@ -64,5 +81,33 @@ goog.exportSymbol('testPcscApi', {
         'SmartCardConnectorApplicationTestHelper', {});
     await pcscReadinessTracker.promise;
   },
+
+  // Test `SCardEstablishContext()`.
+  'testScardEstablishContext': async function() {
+    await testController.setUpCppHelper(
+        'SmartCardConnectorApplicationTestHelper', {});
```
Comment (ViktoriiaKovalova) [inline smart_card_connector_app/src/pcsc-api-jstocxxtest.js:88]: can this be moved to SetUp?

File: smart_card_connector_app/src/pcsc-api-jstocxxtest.js [hunk 2]
```diff
@@ -64,5 +81,33 @@ goog.exportSymbol('testPcscApi', {
...
+        GSC.PcscLiteClient.API.SCARD_SCOPE_SYSTEM);
+    let sCardContext;
+    result.get(
+        (context) => {
+          sCardContext = context;
+        },
+        (errorCode) => {
+          fail(`Unexpected error ${errorCode}`);
+        });
+    assert(Number.isInteger(sCardContext));
```
Comment (ViktoriiaKovalova) [inline smart_card_connector_app/src/pcsc-api-jstocxxtest.js:107]: maybe check if it's equal to SUCCESS?

**Review by ViktoriiaKovalova — APPROVED** [review]
LGTM

File: smart_card_connector_app/src/pcsc-api-jstocxxtest.js (hunk 2 above)
Comment (emaxx-google) [reply smart_card_connector_app/src/pcsc-api-jstocxxtest.js]: Hmm this is a random handle, not an error code. We don't know the exact value to expect here.

**Comment by emaxx-google** [comment]
Refactored it a bit. How do you like it?

**Comment by emaxx-google** [comment]
so lgty? (Github reset it after I asked to re-review)

**Review by ViktoriiaKovalova — APPROVED** [review]

**Pull request merged**